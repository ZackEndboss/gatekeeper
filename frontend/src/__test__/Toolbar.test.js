import React from 'react';
import renderer from 'react-test-renderer';
import {MockStore} from './mockStore';
import {Provider} from 'react-redux';
import ConnectedToolbar, {Toolbar} from '../Toolbar';

describe('Toolbar component', () => {
    it('render', () => {
        const component = renderer.create(<Toolbar openEditGateDialog="openEditGateDialog"/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('Toolbar load from state', () => {
    it('should load the state correctly', () => {
        const storeMock = MockStore({
            'gates': {
                'ftx': {
                    'gatekeeper': {
                        'test1': {},
                        'test2': {}
                    }
                }
            }
        });

        const subject = (
            <Provider store={storeMock}>
                <ConnectedToolbar/>
            </Provider>);
        const component = renderer.create(subject);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});