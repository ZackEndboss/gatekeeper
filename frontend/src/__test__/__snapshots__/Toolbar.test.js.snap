// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar component render 1`] = `
<div
  className="btn-toolbar"
  role="toolbar"
  style={
    Object {
      "margin": "5px",
    }
  }
>
  <button
    className="btn btn-info"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Add Gate
  </button>
  <button
    className="btn btn-info"
    disabled={false}
    type="button"
  >
    Edit Holidays
  </button>
</div>
`;

exports[`Toolbar load from state should load the state correctly 1`] = `
<div
  className="btn-toolbar"
  role="toolbar"
  style={
    Object {
      "margin": "5px",
    }
  }
>
  <button
    className="btn btn-info"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Add Gate
  </button>
  <button
    className="btn btn-info"
    disabled={false}
    type="button"
  >
    Edit Holidays
  </button>
</div>
`;
